### PR TITLE
*: Support limit with DAG

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: store/tikv/mock-tikv/rpc.go [hunk 1]
```diff
@@ -435,18 +435,13 @@ func (c *RPCClient) SendCopReq(ctx goctx.Context, addr string, req *coprocessor.
...
-	}
-
+// extractExecutors is used for testing.
+func extractExecutors(sel *tipb.SelectRequest) []*tipb.Executor {
 	var executors []*tipb.Executor
-	// TODO: Now the value of desc is false.
-	// And it's used for testing.
-	desc := false
+	var desc bool
+	if sel.OrderBy != nil {
```
Comment (hanfei1991) [inline store/tikv/mock-tikv/rpc.go]: Your check is wrong....
it means a desc scan when sel.OrderBy[0].Expr is nil.

File: store/tikv/mock-tikv/rpc.go [hunk 2]
```diff
@@ -435,18 +435,13 @@ func (c *RPCClient) SendCopReq(ctx goctx.Context, addr string, req *coprocessor.
...
-// SendCopReqNew sends a coprocessor request to mock cluster.
-func (c *RPCClient) SendCopReqNew(addr string, req *coprocessor.Request, timeout time.Duration) (*coprocessor.Response, error) {
-	sel := new(tipb.SelectRequest)
-	err := proto.Unmarshal(req.Data, sel)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
+// extractExecutors is used for testing.
+func extractExecutors(sel *tipb.SelectRequest) []*tipb.Executor {
```
Comment (shenli) [inline store/tikv/mock-tikv/rpc.go]: Seems it is not used in unit test.

**Comment by hanfei1991** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**